foreshadow/tests/test_preprocessor.py

```diff
@@ -752,3 +756,17 @@ def test_preprocessor_serialize():
     print(json.dumps(out))
 
     assert json.loads(json.dumps(truth)) == json.loads(json.dumps(out))
+
+
+def test_preprocessor_y_var_filtering():
+    import pandas as pd
+    from foreshadow.preprocessor import Preprocessor
+
+    df = pd.read_csv("./foreshadow/tests/test_data/boston_housing.csv")
+    y_df = df[["medv"]]
+
+    proc = Preprocessor(y_var=True)
+
+    df_out = proc.fit_transform(y_df)
+
+    assert y_df.equals(df_out)
```

Comment:
Why do we assume the input and the output are the same here? For Y preprocessor, shall we always check

```python
df_out = proc.fit_transform(y_df)
assert y_df.equals(proc.inverse_transform(df_out))
```

Reply:
In this case, y_df is the same as the output because in the setup of the test at the top of the file NumericIntent only has one transformer which is the imputation. This code doesn't directly use the IntentTree here because all of the tests might need constant upkeep as intents grow and change